docs/observability.md

```diff
@@ -0,0 +1,149 @@
+
+## Edge-specific Requirements
+
+Capturing signals is done primarily at the edge due to the architecture of this sample. Having only a few components like Event Hubs and Azure Container Registry as current cloud resources, the bigger portion of the observability solution needs to run on th edge.
+
+In addition to the observability stack deployment to measure, analyze and act on monitoring of the platform, the edge comes with some additional challenges. Special network topology with restricted access between network layers, occasionally disconnected scenarios and sometimes need for a local-only or at least locally available copy of the solution to address longer periods of cloud network losses. Local persistence and queuing of data in case of prolonged connectivity loss is an additional complexity.
+
+This solution starts by addressing network topology requirements by building on top of the [nested proxy with Envoy solution](./reverseproxy.md), and aggregate the monitoring solution in the cloud. At the edge there is a local optional observability stack deployed to offer edge based visualization of the signals through Grafana.
+
+## OpenTelemetry
+
+With Microsoft's commitment to embrace OpenTelemetry in its Azure Monitor products - [Azure Monitor Application Insights - OpenTelemetry](https://learn.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-overview#opentelemetry) - this sample is taking a dependency on this open-source, industry supported telemetry instrumentation technology.
+
+OpenTelemetry consists of two functionalities:
+
+- OpenTelemetry Protocol (OTLP)
+- OpenTelemetry Collector
+
+### OpenTelemetry Kubernetes Operator
+
+The OpenTelemetry Operator manages OpenTelemetry Collector and allows for auto-instrumentation of code in different languages. When configuring `receivers` that are standard extensions, the operator ensures ClusterIP services are created automatically within the cluster, minimizing custom setup within the deployment approach with custom Helm charts.
+
+Evaluation is still ongoing whether it adds value in this solution to use the Operator, versus managing the Collector directly. The main reason for not using Operator today is the lack of needs for auto-instrumention injection as all custom workloads leverage Dapr which has its own set of monitoring options already integrated.
+
+OpenTelemetry Collector Deployment Options
+
+- Deployment
+- DaemonSet
+- StatefulSet
+- Sidecar
+
+### Azure Monitor Exporter for OpenTelemetry
+
+This exporter is part of OpenTelemetry-Collector-contrib repo, and is in beta at time of writing.
+
+Details: [Azure Monitor Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter).
+
+Review the documentation to understand how the mapping of attributes works, and in which tables the traces, logs and metrics are stored in Azure Application Insights.
```

Comment:
I would say:
"Refer to the documentation to get a better understanding how...

Reply:
Done